invenio_rdm_records/services/schemas/record.py

```diff
@@ -91,16 +91,12 @@ def default_nested(self, data):
 
         return data
 
-    def hide_tombstone_or_metadata(self, data):
+    def hide_tombstone(self, data):
         """Hide tombstone info if the record isn't deleted and metadata if it is."""
-        is_deleted = data.get("deletion_status", {}).get("is_deleted", False)
+        is_deleted = (data.get("deletion_status") or {}).get("is_deleted", False)
+        tombstone_visible = (data.get("tombstone") or {}).get("is_visible", True)
 
-        if is_deleted:
-            keys_to_keep = ["tombstone", "deletion_status", "id", "pids", "links"]
-            for key in [k for k in data if k in keys_to_keep]:
-                data.pop(key, None)
-
-        else:
```

Comment:
this was a bit overzealous; the actual stripping of data now happens in the rendering (i.e. jinja template for UI, and resource [error handler] for the API)